G4ShareSpy/G4CGMManagerSettingsViewController.swift

```diff
@@ -16,13 +17,27 @@ class G4CGMManagerSettingsViewController: UITableViewController {
 
     public let cgmManager: G4CGMManager
 
-    private var glucoseUnit: HKUnit
+    private let displayGlucoseUnitObservable: DisplayGlucoseUnitObservable
 
-    public init(cgmManager: G4CGMManager, glucoseUnit: HKUnit) {
+    private var cancellable: AnyCancellable?
+
+    private var glucoseUnit: HKUnit {
+        displayGlucoseUnitObservable.displayGlucoseUnit
+    }
+
+    public init(cgmManager: G4CGMManager, displayGlucoseUnitObservable: DisplayGlucoseUnitObservable) {
         self.cgmManager = cgmManager
-        self.glucoseUnit = glucoseUnit
+        self.displayGlucoseUnitObservable = displayGlucoseUnitObservable
 
         super.init(style: .grouped)
+
+        cancellable = displayGlucoseUnitObservable.updatePublisher.sink { [weak self] in
```

Comment:
[nit] Consider above comment, plus:
```
$displayGlucoseUnitObservable
    .sink { [weak self] in self?.tableView.reloadData() }
    .store(in: &cancellables)
```
Not sure if this will work here.